waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -236,7 +230,7 @@ public DatabaseMapping databaseMapping(@NotNull String databaseName) {
   }
 
   private boolean isWhitelisted(String databasePrefix, String database) {
-    List<String> whitelist = whitelistedDbByPrefix.get(databasePrefix);
+    Whitelist whitelist = whitelistedDbByPrefix.get(databasePrefix);
     if (whitelist == null || whitelist.isEmpty()) {
```

Comment:
Just seeing this now, but 
if the whitelist is not configured we accept everything that's ok
if the whitelist is empty we also accept everything isn't that wrong?

Reply:
what happens when WD stops and saves the config does do you end up with a `mapped-databases:[]` even if nothing was there when starting up?

Reply:
```
#Initial config
primary-meta-store:
  access-control-type: READ_ONLY
  database-prefix: ''
  name: primary
  remote-meta-store-uris: thrift://localhost:9083 
federated-meta-stores:
- access-control-type: READ_ONLY
  remote-meta-store-uris: thrift://foo-baz.aws.hcom:48869
  mapped-databases: [] 
  database-prefix: waggle_test_
  name: secondary

#Stop Waggle Dance
[hadoop@ip-10-29-168-174 conf]$ sudo service waggle-dance stop
Stopped [23374]

#Waggle Dance updates config to:
[hadoop@ip-10-29-168-174 conf]$ cat waggle-dance-federation.yml
primary-meta-store:
  access-control-type: READ_ONLY
  database-prefix: ''
  name: primary
  remote-meta-store-uris: thrift://localhost:9083
federated-meta-stores:
- access-control-type: READ_ONLY
  database-prefix: waggle_test_
  name: secondary
  remote-meta-store-uris: thrift://foo-baz.aws.hcom:48869
```

Both variations result in all databases being mapped

Reply:
ok I mean if initial config didn't set `mapped-databases:` but this is good too. I guess I got my answer it won't add an empty list.
Looking at the config I guess it is ok, it doesn't make much sense to have an empty list, you can remove the federated-metastore completely from the config instead. Doesn't make much sense to configure a metastore and not allow access to any databases.
Thanks for checking.